### PR TITLE
Utility function to apply trace line limits

### PR DESCRIPTION
**Pull request by arifBurakDemiray**

### CODE REVIEW

File: sdk/src/main/java/ly/count/android/sdk/UtilsInternalLimits.java [hunk 1]
```diff
@@ -316,6 +316,35 @@ static boolean removeUnsupportedDataTypes(@NonNull Map<String, Object> data) {
...
+        assert tag != null;
+        assert L != null;
+        
+        StringBuilder sb = new StringBuilder(stackTrace.length());
+
+        String[] stackTraceLines = stackTrace.split("\n");
+        for (int i = 0; i < stackTraceLines.length; i++) {
+            String truncatedLine = UtilsInternalLimits.truncateString(stackTraceLines[i], maxStackTraceLineLength, L, tag);
+            if (i != 0) {
+                sb.append("\n");
```
Comment (turtledreams) [inline sdk/src/main/java/ly/count/android/sdk/UtilsInternalLimits.java:340]: if you append this first in the loop wouldnt it make the last line not end with new line?

File: sdk/src/androidTest/java/ly/count/android/sdk/UtilsInternalLimitsTests.java [hunk 2]
```diff
@@ -503,4 +503,27 @@ public void applySdkInternalLimitsToSegmentation_clashingKeys() {
...
+
+    @Test
+    public void applyInternalLimitsToStackTraces() {
+        String stackTrace = "java.lang.Exception: test\n"
+            + "at ly.count.android.sdk.UtilsInternalLimitsTests.applyInternalLimitsToStackTraces(UtilsInternalLimitsTests.java:1)\n"
+            + "at ly.count.android.sdk.UtilsInternalLimitsTests.applyInternalLimitsToStackTraces(UtilsInternalLimitsTests.java:2)\n"
+            + "at ly.count.android.sdk.UtilsInternalLimitsTests.applyInternalLimitsToStackTraces(UtilsInternalLimitsTests.java:3)\n";
+
+        String truncatedStackTrace = UtilsInternalLimits.applyInternalLimitsToStackTraces(stackTrace, 2, "tag", new ModuleLog());
+        Assert.assertEquals("ja\nat\nat\nat", truncatedStackTrace);
```
Comment (turtledreams) [inline sdk/src/androidTest/java/ly/count/android/sdk/UtilsInternalLimitsTests.java:515]: as in here. is there a current example how we send the stacktrace? and how does it look at the server with and without new line for the last line?

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: sdk/src/main/java/ly/count/android/sdk/UtilsInternalLimits.java (hunk 1 above)
Comment (arifBurakDemiray) [reply sdk/src/main/java/ly/count/android/sdk/UtilsInternalLimits.java:340]: yes that is the purpose, last line shouldn't have a newline right?

File: sdk/src/androidTest/java/ly/count/android/sdk/UtilsInternalLimitsTests.java (hunk 2 above)
Comment (arifBurakDemiray) [reply sdk/src/androidTest/java/ly/count/android/sdk/UtilsInternalLimitsTests.java:515]: no let me prep

Comment (arifBurakDemiray) [reply sdk/src/androidTest/java/ly/count/android/sdk/UtilsInternalLimitsTests.java:515]: image
without new line at the end of the last line

Comment (turtledreams) [reply sdk/src/androidTest/java/ly/count/android/sdk/UtilsInternalLimitsTests.java:515]: its cool then

File: sdk/src/main/java/ly/count/android/sdk/UtilsInternalLimits.java (hunk 1 above)
Comment (turtledreams) [reply sdk/src/main/java/ly/count/android/sdk/UtilsInternalLimits.java:340]: 🤷

**Review by turtledreams — APPROVED** [review]

**Review by AliRKat — APPROVED** [review]

**Pull request merged**